src/devices/Common/System/Device/Analog/AnalogController.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+
+namespace System.Device.Analog
+{
+    /// <summary>
+    /// Base class for Analog Controllers.
+    /// These control analog input pins.
+    /// </summary>
+    public abstract class AnalogController : IDisposable
+    {
+        private readonly List<AnalogInputPin> _openPins;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GpioController"/> class that will use the specified numbering scheme and driver.
+        /// </summary>
+        /// <param name="numberingScheme">The numbering scheme used to represent pins provided by the controller.</param>
+        public AnalogController(PinNumberingScheme numberingScheme)
```

Comment:
consider making this protected

Reply:
although here I think it doesn't make any difference - match visiblility with other similar classes in .NET

Reply:
Class is abstract, so a public ctor is useless - changed.